modules/ingestor/src/graph/sbom/cyclonedx.rs

```diff
@@ -91,16 +91,23 @@ impl SbomContext {
     #[instrument(skip(tx, sbom), ret)]
     pub async fn ingest_cyclonedx<TX: AsRef<Transactional>>(
         &self,
-        sbom: Bom,
+        mut sbom: Bom,
         tx: TX,
     ) -> Result<(), anyhow::Error> {
         let db = &self.graph.db.connection(&tx);
 
         let mut license_creator = LicenseCreator::new();
         let mut creator = Creator::new(self.sbom.sbom_id);
 
-        if let Some(metadata) = &sbom.metadata {
-            if let Some(component) = &metadata.component {
+        // extract "describes"
+
+        if let Some(metadata) = &mut sbom.metadata {
+            if let Some(component) = &mut metadata.component {
+                let bom_ref = component
+                    .bom_ref
+                    .get_or_insert_with(|| Uuid::new_v4().to_string())
```

Comment:
we generate a uuid if no bom_ref exists ?

Reply:
Yes, we need to link it in the database. As the database uses references, where the cyclonedx uses a "contains" model to describe this information.